cmd/livenessprobe/main.go

```diff
@@ -119,16 +125,37 @@ func acquireConnection(ctx context.Context, metricsManager metrics.CSIMetricsMan
 }
 
 func main() {
-	klog.InitFlags(nil)
+	var fs flag.FlagSet
+	klog.InitFlags(&fs)
+	fs.VisitAll(func(f *flag.Flag) {
+		switch f.Name {
+		case "logtostderr":
+			flag.CommandLine.Var(f.Value, f.Name, f.Usage)
+		}
+	})
+	fg := featuregate.NewFeatureGate()
+	logsapi.AddFeatureGates(fg)
+	c := logsapi.NewLoggingConfiguration()
+	logsapi.AddGoFlags(c, flag.CommandLine)
+	logs.InitLogs()
 	flag.Set("logtostderr", "true")
 	flag.Parse()
+	if err := logsapi.ValidateAndApply(c, fg); err != nil {
+		klog.ErrorS(err, "LoggingConfiguration is invalid")
+		os.Exit(1)
+	}
+	if c.Format == logsapi.JSONLogFormat {
+		logger, _ := json.NewJSONLogger(c.Verbosity, os.Stderr, nil, nil)
+		klog.SetLogger(logger)
+		defer klog.ClearLogger()
+	}
```

Comment:
This looks like quite a big chunk of code that could live in [csi-lib-utils](https://github.com/kubernetes-csi/csi-lib-utils), so all sidecars can benefit from it. BTW, is there anything in `component-base` that we could use to set up logging at a sidecar startup?

Reply:
I apologize for the oversight. On double-checking, I realized that it wasn't necessary to manually reset the log format, so I've fixed the code.

> This looks like a fairly large chunk of code that could live in csi-lib-utils so that all sidecars can benefit from it.

Indeed, even after the changes, the code is about 20 lines long. It might be a good idea to move this implementation to a function within csi-lib-utils.